epoxy-adapter/src/main/java/com/airbnb/epoxy/VisibilityState.java

```diff
@@ -45,4 +45,12 @@
    * become visible.
    */
   public static final int FULL_IMPRESSION_VISIBLE = 4;
+
+  /**
+   * Event triggered when a Component enters the Partial Impression Range. This happens, for
+   * instance in the case of a vertical RecyclerView, when the percentage of the visible area is
+   * greater than a specified threshold. The threshold can be set in
+   * {@link EpoxyVisibilityTracker#setPartialImpressionThreshold(int)}.
+   */
+  public static final int PARTIAL_IMPRESSION_VISIBLE = 5;
```

Comment:
If you look at the `FOCUSED_VISIBLE` it protect against component that are bigger than the view port. 

```
  /**
   * Event triggered when a Component enters the Focused Range. This happens when either the
   * Component occupies at least half of the viewport or, if the Component is smaller than half of
   * the viewport, when the it is fully visible.
   */
  public static final int FOCUSED_VISIBLE = 2;
```

I was wondering if you want to implements the same protection?

The problem is, if one of the component is bigger than view port (or if threshold % bigger that view port) you will never receive a `PARTIAL_IMPRESSION_VISIBLE` event.

Reply:
I'll have a look into `FOCUSED_VISIBLE` and it's protection

Reply:
> The problem is, if one of the component is bigger than view port (or if threshold % bigger that view port) you will never receive a PARTIAL_IMPRESSION_VISIBLE event.

What do you think users would expect? I see some possible ways of dealing with this:

1) Do actually define it as accurate behavior that no event is fired if the threshold cannot be met. The events would more likely to be fired if the threshold is small. Event firing cannot be guaranteed though as you mentioned.
2) If the element is bigger than the viewport, change the frame of reference for the threshold to relate to the viewport instead of the element (so with a 50% threshold the event would fire once the element covers 50% of the viewport). The question here is if this two-fold behavior could be confusing to users of the lib or just plain undesired.
3) Hybrid approach: For as long as the threshold can be met, fire it. In the case the threshold cannot be reached, fired it if the view covers the whole viewport. 

@eboudrant From your experience with epoxy in use and feedback so far, which of these behaviors (or which other suggestion) would be most appropriate?

Reply:
Another comment: I looked at the code for `FULL_IMPRESSION_VISIBLE`, which also does not handle the case if an element is bigger than the viewport. I tend towards keeping this same behavior for the `PARTIAL_IMPRESSION_VISIBLE` as well.